### DISCUSSION TITLE
Failed to install Homebrew Portable Ruby

### DISCUSSION THREAD

**Discussion by ddlzr** [Homebrew/discussions#2658] [Everyday usage] [answered]
Today upgraded macos to 11.6.2 from 11.6.1, then  upgraded "Command Line Tools For Xcode" to 13.2 from 13.0, after excuted "rm -rf /usr/local/Homebrew/Library/Homebrew/vender" command, can not reinstall portable-ruby-2.6.8.

![image](https://user-images.githubusercontent.com/22073788/146717315-a94cf039-f266-41b2-9b5f-729bb2016629.png)

![image](https://user-images.githubusercontent.com/22073788/146717655-7c39e317-4f1e-4fab-bebb-0a2abedca017.png)

**Comment by ddlzr** [comment]
The command  also dosen't work.
`git -C /usr/local/Homebrew clean -dfx`

**Comment by SMillerDev** [comment]
> after excuted "rm -rf /usr/local/Homebrew/Library/Homebrew/vender" command, can not reinstall portable-ruby-2.6.8.

What made you run this?

Additionally, what do `brew config` and `brew doctor` say?

**Reply by ddlzr** [reply]
I thought  "rm -rf /usr/local/Homebrew/Library/Homebrew/vender"  is  equivalent to 'git -C /usr/local/Homebrew clean -dfx'. I want to clean  the BREW cache after upgraded System  And CLT version.

brew config details :
```
[huwang@huwangdeMacBook-Pro]~ $ brew config
cat: /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby-version: No such file or directory
cat: /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby-version: No such file or directory
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:0cb1cc7af109437fe0e020c9f3b7b95c3c709b140bde9f991ad2c1433496dd42
############################################################################################################################################################################################################# 100.0%
==> Pouring portable-ruby-2.6.8.yosemite.bottle.tar.gz
Error: Failed to install ruby !
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
```
brew doctor details:
```
[huwang@huwangdeMacBook-Pro]~ $ brew doctor
cat: /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby-version: No such file or directory
cat: /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby-version: No such file or directory
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:0cb1cc7af109437fe0e020c9f3b7b95c3c709b140bde9f991ad2c1433496dd42
Already downloaded: /Users/huwang/Library/Caches/Homebrew/portable-ruby-2.6.8.yosemite.bottle.tar.gz
==> Pouring portable-ruby-2.6.8.yosemite.bottle.tar.gz
Error: Failed to install ruby !
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
```

**Reply by SMillerDev** [reply]
Neither of those clean the brew cache, `brew cleanup --prune=all` will clear your cache. Additionally, there is no need to clear your cache because there is no difference in the builds. Only going from macOS `11 -> 12` might need a cache cleanup and even that isn't required.

---

As for your issue, it's likely the GFW blocking or hampering the download of portable ruby. You can check https://mirrors.tuna.tsinghua.edu.cn/help/homebrew/ to see if they have some solutions to this.

**Reply by ddlzr** [reply]
After running  "brew config" , "portable-ruby-2.6.8.yosemite.bottle.tar.gz" had downloaded  with VPN, sha256 verification also  passed. However, I have no idea which caused "Failed to install Homebrew Portable Ruby (and your system version is too old)" .

**Reply by SMillerDev** [reply]
Try running `brew config --debug` to see if it prints more information

**Reply by ddlzr** [reply]
There was  no more info. To re-download, I removed file by "rm -rf /Users/huwang/Library/Caches/Homebrew/portable-ruby-2.6.8.yosemite.bottle.tar.gz".
```
[huwang@huwangdeMacBook-Pro]~ $ brew config --debug
cat: /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby-version: No such file or directory
cat: /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby-version: No such file or directory
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:0cb1cc7af109437fe0e020c9f3b7b95c3c709b140bde9f991ad2c1433496dd42
############################################################################################################################################################################################################# 100.0%
==> Pouring portable-ruby-2.6.8.yosemite.bottle.tar.gz
Error: Failed to install ruby !
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
```

**Reply by SMillerDev** [reply]
Tagging in @Homebrew/brew to see if there is anything they can help with.

**Reply by ddlzr** [reply]
Thank you for your help.

**Comment by carlocab** [comment]
Deleting the `vendor` directory is almost certainly going to break `portable-ruby` installations, since `brew` uses `vendor/portable-ruby-version` to work out which `portable-ruby` it should be installing. Try doing
```
mkdir -p /usr/local/Homebrew/Library/Homebrew/vendor
echo "2.6.8" >> /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby-version
```
and see if it helps.

**Reply by ddlzr** [reply]
Thanks for your help. After the commands shown above. Running `brew doctor `or other `brew` command  will cause some error as  shown below.  It solved by `brew update-reset` which  reseted  flies that I deleted.
![image](https://user-images.githubusercontent.com/22073788/146745026-11024b2f-6550-4878-9eae-5f6cc33b4f60.png)

**Comment by MikeMcQuaid** [comment, marked as answer]
Run `brew update-reset` and try again.

**Reply by ddlzr** [reply]
Fixed. Thanks for your help.
It's my vault, I thought `rm -rf /usr/local/Homebrew/Library/Homebrew/vendor` was equal to `git -C /usr/local/Homebrew clean -dfx`. However, the `vender` directory and some rescusive files are tracked by git. brew `update-reset` reset files that removed. Running `brew config` or other `brew` command, portable ruby  installed automatically.
